server/src/bin/ingestion-worker.rs

```diff
@@ -416,6 +419,34 @@ pub async fn bulk_upload_chunks(
         "precomputing some important data before insert",
     );
 
+    let unlimited = std::env::var("UNLIMITED").unwrap_or("false".to_string());
+    if unlimited == "false" {
+        let dataset_org_plan_sub = get_dataset_and_organization_from_dataset_id_query(
+            models::UnifiedId::TrieveUuid(payload.dataset_id),
+            None,
+            web_pool.clone(),
+        )
+        .await?;
+
+        let chunk_count = get_row_count_for_organization_id_query(
+            dataset_org_plan_sub.organization.organization.id,
+            web_pool.clone(),
+        )
+        .await?;
+
+        if chunk_count + payload.ingestion_messages.len()
+            > dataset_org_plan_sub
+                .organization
+                .plan
+                .unwrap_or_default()
+                .chunk_count as usize
+        {
+            return Err(ServiceError::BadRequest(
+                "Chunk count exceeds plan limit".to_string(),
+            ));
+        }
+    }
+
```

Comment:
Why is this getting added in this PR? Were we missing it or was it not being applied ever?

Reply:
We were missing it in ingestion worker, this is why we were allowing ppl to have more chubks than they were allowed for site search etc